libexec/rtld-elf/rtld_paths.h

```diff
@@ -45,16 +45,22 @@
 #endif
 #endif
 
-#ifndef _RTLD_C18N_FILE_SUFFIX
```

Comment:
My comment for all of these:

What is the intended set of files used by each linker in the various different scenarios? I foresee issues with ld-elf being both c18n and non-c18n when it comes to ldconfig, since you need the c18n libc+libthr for c18n but the non-c18n one for non-c18n, yet it's the same rtld. What's making sure the right one gets used? Similarly what's making sure the files are configured correctly by /etc/rc.d/ldconfig? Note that these days it asks rtld itself for the system lib path, which may give different output based on what the default c18n configuration is, which sounds like something that you don't want.